simphony/testing/abc_check_engine.py

```diff
@@ -25,6 +30,11 @@ def setUp(self):
         self.items = self.create_dataset_items()
 
     @abc.abstractmethod
+    def container_factory(self, name, mode='w'):
```

Comment:
The `abc_check_engine.py` is to be used for testing by lammps-md and kratos which follow the ABCModelingEngine class but both don't have a concept of file mode.

I think the confusion is that some of the tests are originally from testing H5CUDS and are testing specific things related to files.  For example, in the current `test_add_get_dataset()` which was from here (lines from version 0.1.5:   https://github.com/simphony/simphony-common/blob/0.1.5/simphony/io/tests/test_h5_cuds.py#L158-161), there is testing related to the behaviour of a H5Particles instance  and the H5Cuds file when the H5CUDS file is closed.  The tests are ensuring that exceptions are thrown when the file has been close (i.e. the user can't use a closed file and the user can no longer use the H5Particles instance which was from that closed file).

Maybe these file-specific tests should only be used in the testing of H5Cuds (i.e. only appear in 'test_h5cuds.py').  The general tests of ABCModelingEngine should be in 'testing/abc_check_engine.py'.